src/main/java/com/fasterxml/jackson/databind/deser/ThreadLocalBuilderBasedDeserializer.java

```diff
@@ -221,6 +221,10 @@ private Object vanillaDeserializeAndFinishBuild(
                     });
         } catch (final WrappedIOExceptionException e) {
             throw e.getCause();
+            // CHECKSTYLE.OFF: IllegalCatch - Match behavior in BuilderBasedDeserializer
+        } catch (final Exception e) {
+            // CHECKSTYLE.ON: IllegalCatch
+            return _underlyingBuilderBasedDeserializer.wrapInstantiationProblem(e, ctxt);
```

Comment:
I'm a bit confused.  The block before this, it looks like you're unwrapping the exception.  But here, it looks like you're wrapping an exception and returning it?  Should this be a throw?  Should it be wrapped?

Reply:
The previous block deals with caught IOExceptions which need to be wrapped in order to escape the builder `Consumer`. These are unwrapped and thrown as-is. The second block deals with runtime exceptions and invokes the same method that `BuilderBaseDeserializer` does. My only worry is that they separate construction and population, so it's possible we may be incorrectly mapping some exceptions.